### PR TITLE
Default material color for 3mf loader

### PR DESCRIPTION
**Pull request by kovacsv**
3mf files often come without material definitions. In this case the default material color was gratuitously purple. I've added the possibility to set this default color from the outside.

### CODE REVIEW

File: examples/jsm/loaders/3MFLoader.js [hunk 1]
```diff
@@ -47,9 +47,16 @@ class ThreeMFLoader extends Loader {
 		super( manager );
 
 		this.availableExtensions = [];
+		this.defaultMaterialColor = 0xaaaaff;
```
Comment (Mugen87) [inline examples/jsm/loaders/3MFLoader.js:50]: Can we define this as a `THREE.Color` object? `setDefaultMaterialColor()` would be defined like so then:

> this.defaultMaterialColor.set( defaultMaterialColor );

It's more consistent if color properties are actually of type `THREE.Color` like with materials.

**Comment by Mugen87** [comment]
Um, in general we do not provide loader interfaces that allow to configure default setups like material colors or material types etc. If we start with `3MFLoader`, we potentially have to do this for other loaders, too.

@mrdoob What is your preference?

File: examples/jsm/loaders/3MFLoader.js (hunk 1 above)
Comment (kovacsv) [reply examples/jsm/loaders/3MFLoader.js]: Good point, I've modified it to store a `THREE.Color` object.

**Comment by kovacsv** [comment]
Just wanted to add, that there are two types of loader parameters:
1. Loader independent, like `setDefaultMaterialColor`. It can be utilized by almost all loaders, and could be placed in the `Loader` base class, although it can be confusing if not all loaders utilize this value.
2. Loader specific parameters, like turning off the huge sphere generation for sky in the `VRMLLoader`. This one probably should be loader dependent.

**Comment by kovacsv** [comment]
> It's actually no big deal to change the material color on app level after the loading process.

Yes, if you know which materials are generated by the loader, and which come from the model. Usually it's not easy to tell. Even if you understand the internals of the loader you can't decide if it's the default material, or a material with the same color that is coming from the model.

**Comment by gkjohnson** [comment]
Is there a good reason for the default color to be 0xaaaaff? I would just expect the default color to be white like other loaders do (OBJLoader, for example).

**Comment by mrdoob** [comment]
I agree. I think it would be better to just make `0xffffff` the default. 

@kovacsv Would you still need something like `setDefaultMaterialColor()` in that case?

**Comment by kovacsv** [comment]
@mrdoob On the long term I'd be happy to see a common interface for all loaders to set the default material, because in my use case users are able to modify the material if the imported model doesn't contain any material information.

On the other hand I understand that this is a bigger task, and for now changing the 3mf default color to white seems better than the purple one, so I'll open another pull request to do that.

Thanks for the help and the input, I close this one now.

**Pull request closed without merging**